client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,271 @@
+  recordingMinutes?: string;
+  /** Filename of recording, if recording */
+  recordingFile?: string;
+  /** Remaining disk space available for recording */
+  diskSpaceRemaining?: string;
+  /** Description of any error that has occurred */
+  error?: string;
+  /** Catch-all for any extra data */
+  [propName: string]: any;
+}
+
+/**
+ * Parse the recording payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param data Payload from MQTT message, parsed
+ * @returns Formatted payload
+ */
+function parseRecordingPayloadData(data: CameraRecordingStatusPayload) {
+  if (!data) return null;
+
+  const formattedData: CameraRecordingStatusFormatted = {};
+
+  Object.keys(data).forEach((field) => {
+    let name = camelCaseToStartCase(field);
+    let value = '';
+
+    // format field value
+    switch (field) {
+      case 'status':
+        value = capitalise(data.status);
+        break;
+
+      case 'diskSpaceRemaining':
+        value = formatBytes(data.diskSpaceRemaining);
+        break;
+
+      case 'recordingMinutes':
+        {
+          const totalMinutes = data.recordingMinutes as number;
+          const mins = Math.floor(totalMinutes);
+          const secs = Math.floor((totalMinutes - mins) * 60);
+          value = `${mins}m ${secs}s`;
+          name = 'Recording Time';
+        }
+        break;
+
+      default:
+        value = data[field];
+        break;
+    }
+
+    formattedData[name] = value;
+  });
+
+  return formattedData;
+}
+
+/**
+ * Initiate receiving status payloads
+ *
+ * @param subComponent The sub component to init for (e.g. recording, video feed)
+ * @param device Device
+ */
+function initStatus(subComponent: string, device?: CameraDevice) {
+  const deviceAsArr = device ? [device] : []; // only add device path if specified
+  const path = ['camera', subComponent].concat(deviceAsArr);
+  emit(`get-status-payload`, path);
+}
+
+/**
+ * Makes the device name pretty.
+ *
+ * Hardcoded for efficiency. If you don't like it, complain to Angus Trau :).
+ *
+ * @param device Device
+ * @returns  Prettied device name
+ */
+export function getPrettyDeviceName(device: CameraDevice) {
+  return device === CameraDevice.Primary ? 'Primary' : 'Secondary';
+}
+
+interface StatusPayloadOptions<T> {
+  /** Initial value for the payload */
+  initValue?: T;
+  /** Handler for updating payload */
+  payloadHandler?: (
+    setter: (p?: T) => void,
+    newPayload: T,
+    device: CameraDevice,
+  ) => T | void;
+  /** Handler for return value */
+  returnHandler?: (payload?: T, device?: CameraDevice) => T | null | undefined;
+}
+
+/**
+ * Create hooks for getting payloads
+ *
+ * @param sub Subcomponent
+ * @param opts Options
+ * @returns Hook for getting a payload
+ */
+function createStatusPayloadHook<T>(
+  sub: string,
+  {
+    initValue = undefined,
```

Comment:
I think I was drawn to using undefined because it integrates nicely with typescript's `?` syntax, but actually things are so much neater without the mix of undefined and null.